CesiumUtility/test/TestCreditReferencer.cpp

```diff
@@ -0,0 +1,122 @@
+
+    pReferencer->addCreditReference(credit1);
+    pReferencer->addCreditReference(credit2);
+
+    // A second reference to credit1
+    pReferencer->addCreditReference(credit1);
+
+    const CreditsSnapshot& snapshot1 = pCreditSystem->getSnapshot();
+    REQUIRE(snapshot1.currentCredits.size() == 2);
+
+    SUBCASE("and removes them in releaseAllReferences") {
+      pReferencer->releaseAllReferences();
+      const CreditsSnapshot& snapshot2 = pCreditSystem->getSnapshot();
+      REQUIRE(snapshot2.currentCredits.size() == 0);
+    }
+
+    SUBCASE("and removes them in the destructor") {
+      pReferencer.reset();
+      const CreditsSnapshot& snapshot2 = pCreditSystem->getSnapshot();
+      REQUIRE(snapshot2.currentCredits.size() == 0);
+    }
+
+    SUBCASE("and duplicates them in the copy constructor") {
+      CreditReferencer referencer2(*pReferencer);
+
+      pReferencer.reset();
+      const CreditsSnapshot& snapshot2 = pCreditSystem->getSnapshot();
+      REQUIRE(snapshot2.currentCredits.size() == 2);
+
+      referencer2.releaseAllReferences();
+      const CreditsSnapshot& snapshot3 = pCreditSystem->getSnapshot();
+      REQUIRE(snapshot3.currentCredits.size() == 0);
+    }
+
+    SUBCASE("and moves them in the move constructor") {
+      CreditReferencer referencer2(std::move(*pReferencer));
+
+      const CreditsSnapshot& snapshot2 = pCreditSystem->getSnapshot();
+      REQUIRE(snapshot2.currentCredits.size() == 2);
+
+      referencer2.releaseAllReferences();
+      const CreditsSnapshot& snapshot3 = pCreditSystem->getSnapshot();
+      REQUIRE(snapshot3.currentCredits.size() == 0);
+    }
+
+    SUBCASE("and clears them on copy assignment") {
+      CreditReferencer referencer2(pCreditSystem);
+      referencer2.addCreditReference(credit3);
+
+      const CreditsSnapshot& snapshot2 = pCreditSystem->getSnapshot();
+      REQUIRE(snapshot2.currentCredits.size() == 3);
+
+      referencer2 = *pReferencer;
+      const CreditsSnapshot& snapshot3 = pCreditSystem->getSnapshot();
+      REQUIRE(snapshot3.currentCredits.size() == 2);
+
+      pReferencer.reset();
+      const CreditsSnapshot& snapshot4 = pCreditSystem->getSnapshot();
+      REQUIRE(snapshot4.currentCredits.size() == 2);
+
+      referencer2 = CreditReferencer();
+      const CreditsSnapshot& snapshot5 = pCreditSystem->getSnapshot();
+      REQUIRE(snapshot5.currentCredits.size() == 0);
+    }
+
+    SUBCASE("and clears them on move assignment") {
+      CreditReferencer referencer2(pCreditSystem);
+      referencer2.addCreditReference(credit3);
+
+      const CreditsSnapshot& snapshot2 = pCreditSystem->getSnapshot();
+      REQUIRE(snapshot2.currentCredits.size() == 3);
+
+      referencer2 = std::move(*pReferencer);
+      const CreditsSnapshot& snapshot3 = pCreditSystem->getSnapshot();
+      REQUIRE(snapshot3.currentCredits.size() == 2);
+
+      const CreditsSnapshot& snapshot4 = pCreditSystem->getSnapshot();
+      REQUIRE(snapshot4.currentCredits.size() == 2);
```

Comment:
Remove? (unless we're purposely testing snapshots that are consecutive with no changes)